packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -50,12 +66,28 @@ export interface StateProps {
   readonly stateName: string;
 
   /**
-   * Specifies the events on enter. the conditions of the events are evaluated when the state is entered.
-   * If the condition is `TRUE`, the actions of the event are performed.
+   * Specifies the events on enter. The conditions of the events will be evaluated when entering this state.
+   * If the condition of the event evaluates to `true`, the actions of the event will be executed.
    *
-   * @default - events on enter will not be set
+   * @default - no events will trigger on entering this state
    */
   readonly onEnter?: Event[];
+
+  /**
+   * Specifies the events on input. The conditions of the events will be evaluated when any input is received.
+   * If the condition of the event evaluates to `true`, the actions of the event will be executed.
+   *
+   * @default - no events will trigger on input in this state
+   */
+  readonly onInput?: Event[];
+
+  /**
+   * Specifies the events on exit. The conditions of the events are evaluated when an exiting this state.
+   * If the condition evaluates to `true`, the actions of the event will be executed.
+   *
+   * @default - no events will trigger on exiting this state
+   */
+  readonly onExit?: Event[];
```

Comment:
Can we wait with adding this to the next PR? Is this related to Actions at all?

Reply:
Yes we can.
I added this feature because it was useful for testing actions. But now this PR has gotten a little too big, so let's remove it.